### PR TITLE
add_by_email() --> Strip "Fwd: " from subject lines

### PR DESCRIPTION
**Pull request by sharedphysics**
- Removed "Fwd: " from subject lines. This means adding articles as fwded emails will look cleaner. 
- Optional next steps: 
--- Modify re function to account for all different types of "FWD/fwd/Fwd"
--- Modify re function to account for "Re:" (and variations)
--- Process data above the email metadata as a custom note. (I.e., if someone fwds/replies to add to Lurnby, keep the reply/fwd text above the metadata as a note, external to the article.

### CODE REVIEW

File: app/main/routes.py [hunk 1]
```diff
@@ -444,7 +444,10 @@ def add_by_email():
                 content = text
 
             if subject:
-                title = subject
+                # Strip "Fwd: " from subject lines
+                stripped_subject = re.sub("Fwd: ", "", subject)
```
[the remark this answers is not in the document]
Comment (Roznoshchik) [reply app/main/routes.py:448]: Maybe it's even this simple?
```
print (re.sub('bush', 'xxxx', 'Bushmeat', flags=re.IGNORECASE))
```

[the remark this answers is not in the document]
Comment (sharedphysics) [reply app/main/routes.py]: I added in `flags=re.IGNORECASE)`

I'm not sure how to write and embed unit tests, but here is a working example of some cases:

```
import re

test1 = "Fwd: Sample email"
test2 = "Fwd: Sample email about fwding development"
test3 = "fwd: Sample email"
test4 = "fwd: fwd: Sample email"
test5 = "Re: fwd: Sample email"
test6 = "wd: Sample email"
test7 = "fwd Sample email"
test8 = "Fwd:Sample email"

test_group = {test1, test2, test3, test4, test5, test6, test7, test8}

for subject in test_group:
    stripped_subject = re.sub("Fwd: ", "", subject, flags=re.IGNORECASE)
    print(subject," --> ",stripped_subject)
```

Output:
```
Fwd: Sample email about fwding development  -->  Sample email about fwding development # Pass
Re: fwd: Sample email  -->  Re: Sample email # Pass
Fwd:Sample email  -->  Fwd:Sample email # Fail b/c of whitespace.
fwd: Sample email  -->  Sample email # Pass
wd: Sample email  -->  wd: Sample email #Pass
fwd: fwd: Sample email  -->  Sample email # Pass
fwd Sample email  -->  fwd Sample email # Pass (technically)
Fwd: Sample email  -->  Sample email # Pass
```

For the whitespace issue, I''m not sure if you're already doing it so I'm not adding the code in. But I'd say you can safely change to regex out `Fwd:` with no space and then run a whitespace trimmer. The part I'm unsure about is if you're running it somewhere else already.

[the remark this answers is not in the document]
Comment (Roznoshchik) [reply app/main/routes.py]: I wasn't paying attention to the whitespace. I think go ahead and ignore the whitespace and seems like this is working.

You wouldn't embed unit tests into here, you would have them in a tests folder, but that would only work if you made this a function that could be imported and tested.

If this is working then go ahead and push a new commit with these changes and I will approve :)

**Review by Roznoshchik — APPROVED** [review]

File: app/main/routes.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Roznoshchik) [reply app/main/routes.py]: oh. I see you did. It's merged.

**Pull request merged**